src/pydocstyle/checker.py

```diff
@@ -997,9 +997,23 @@ def get_leading_words(line):
         return result.group()
 
 
-def get_function_args(function_string):
+def get_function_args(function_definition):
     """Return the function arguments given the source-code string."""
-    function_arg_node = ast.parse(textwrap.dedent(function_string)).body[0].args
+    function_source = function_definition.source
+    function_docs = function_definition.docstring
+    blank_docstring = "''''''"
+    # We are replacing the docstring with a blank docstring.
+    # This is so that if the docstring has incorrectly
+    # indented lines, which are at a lower indent than the
+    # function source, we still dedent the source correctly
+    # and the AST parser doesn't throw an error.
+    sanitized_source = function_source.replace(function_docs, blank_docstring)
+    try:
+        function_arg_node = ast.parse(textwrap.dedent(sanitized_source)).body[0].args
```

Comment:
I know that we already used `ast.parse` before this change, but I wonder if this works for closures? e.g.,

```.py
def foo():
    x = 1
    def bar():
        """Will this work when referencing x?"""
        print(x)
```

Can you add / make sure we have a test for this?